recipes/doxygen/all/conanfile.py

```diff
@@ -54,10 +54,13 @@ def requirements(self):
             self.requires("xapian-core/1.4.19")
             self.requires("zlib/1.2.13")
 
-    def package_id(self):
-        del self.info.settings.compiler
```

Comment:
This is the primary purpose of this PR. This kind of statement should be removed from all `tool_requires` packages. See option (3) proposed in https://github.com/conan-io/conan-center-index/issues/17034 for more detail

Reply:
I don't understand, if it should be removed, and if it's a primary purpose of this PR, then why it's not removed?
P.S. all the side changes could go into another PR. it's good practice to keep PRs atomic and focused on one thing at time. it's easier to review/approve changes if the do only one thing.

Reply:
> I don't understand, if it should be removed, and if it's a primary purpose of this PR, then why it's not removed?


Yeah, I can appreciate this is confusing. The evolution of the review is such that this wasn't identified as the best approach to resolving the glibcxx problems. I called this out here to distinguish it from the `compatibility()` implementation which was just a complementary proposal in addition to this. What this turned into is statically compiling glibcxx.

> P.S. all the side changes could go into another PR. it's good practice to keep PRs atomic and focused on one thing at time. it's easier to review/approve changes if the do only one thing.

The side changes only became necessary in this PR to get the pipeline working again. Honestly though, given how long it takes to go through a review cycle I'm loathe to split it out as I want to get these changes in. If team review was more responsive id be more inclined to submit more atomic PRs

Reply:
the best approach I could recommend is to create multiple PRs, each one for its own set of changes, e.g.:
- PR 1 - solution 1 (remove `del self.info.settings.compiler`)
- PR 2 - solution 2 (add `compatibility()` method)
- PR 3 - solution 3 (link `libstdc++` statically)
- PR 4 - side changes (e.g. libiconv changes)

then it would be much more easier to read and review every solution. right now, I am sorry to say, but PR is a bit chaotic and hard to review, as PR title, comments in code and comments in review do not match each other, confusing reviewers on what's actually happening here (like comment here says "X" is the main thing in PR, but PR doesn't include "X", and instead includes some unrelated "Y" and "Z", which is mind-blowing).